h/services/group_members.py

```diff
@@ -104,15 +104,14 @@ def update_members(self, group, userids):
                         be the members of this group
         """
         current_mem_ids = [member.userid for member in group.members]
-        userids_for_removal = [
-            mem_id for mem_id in current_mem_ids if mem_id not in userids
-        ]
 
         for userid in userids:
-            self.member_join(group, userid)
+            if userid not in current_mem_ids:
+                self.member_join(group, userid)
 
-        for userid in userids_for_removal:
-            self.member_leave(group, userid)
+        for userid in current_mem_ids:
+            if userid not in userids:
+                self.member_leave(group, userid)
```

Comment:
I also rewrote the code for _removing_ members to use this loop rather than creating the local variable `userids_for_removal`. This just makes the code style consistent with my loop for _adding_ members above, and avoids unnecessary local variables.